### PR TITLE
Personal Info api

### PR DESCRIPTION
**Pull request by pisarevkir**

### CODE REVIEW

File: src/utils/country.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export const getCountryByCode = (countryCode?: string | null) => {
+  return {
+    current: countryCode
+      ? countries.find(
+          country =>
+            country.isoCode.toLowerCase() === countryCode.toLowerCase(),
+        )
+      : null,
+    default: countries[0],
```
Comment (victor-oancea-leftclick-io) [inline src/utils/country.ts:13]: Whats this for?

File: src/screens/UserRegistrationFlow/SignIn/index.tsx [hunk 2]
```diff
@@ -43,10 +42,13 @@ type Props = {
...
 
 export const SignIn = ({navigation}: Props) => {
-  const deviceCountry = useSelector(deviceCountrySelector);
+  const deviceLocation = useSelector(deviceLocationSelector);
+  const deviceCountry = getCountryByCode(deviceLocation?.country);
   const [email, onChangeEmail] = useState('');
   const [phone, setPhone] = useState('');
-  const [selectedCountry, setSelectedCountry] = useState(deviceCountry);
+  const [selectedCountry, setSelectedCountry] = useState(
+    deviceCountry.current ?? deviceCountry.default,
```
Comment (victor-oancea-leftclick-io) [inline src/screens/UserRegistrationFlow/SignIn/index.tsx:50]: Whats the point of that? You default to the one from device location anyway

File: src/screens/SettingsFlow/PersonalInformation/index.tsx [hunk 3]
```diff
@@ -39,18 +40,21 @@ export const PersonalInformation = memo(() => {
...
     useNavigation<NativeStackNavigationProp<ProfileTabStackParamList>>();
 
-  const deviceCountry = useSelector(deviceCountrySelector);
+  const user = useSelector(userSelector);
+  const userCountry = getCountryByCode(user?.country);
 
   const [hasChanges, setHasChanges] = useState(false);
-  const [selectedCountry, setSelectedCountry] = useState(deviceCountry);
+  const [selectedCountry, setSelectedCountry] = useState(
+    userCountry.current ?? userCountry.default,
```
Comment (victor-oancea-leftclick-io) [inline src/screens/SettingsFlow/PersonalInformation/index.tsx:48]: The default thing is useless

File: src/components/ModifyPhoneNumber/index.tsx [hunk 4]
```diff
@@ -26,11 +27,14 @@ export function ModifyPhoneNumber({
...
 }: Props) {
-  const deviceCountry = useSelector(deviceCountrySelector);
+  const deviceLocation = useSelector(deviceLocationSelector);
+  const deviceCountry = getCountryByCode(deviceLocation?.country);
 
   const [phone, setPhone] = useState('');
   const [focused, setFocused] = useState(false);
-  const [selectedCountry, setSelectedCountry] = useState(deviceCountry);
+  const [selectedCountry, setSelectedCountry] = useState(
+    deviceCountry.current ?? deviceCountry.default,
```
Comment (victor-oancea-leftclick-io) [inline src/components/ModifyPhoneNumber/index.tsx:36]: Default is useless

**Review by victor-oancea-leftclick-io — CHANGES_REQUESTED** [review]
user.country will always be set, if user is registered; so remove that default thing

File: src/screens/UserRegistrationFlow/SignIn/index.tsx [hunk 5]
```diff
@@ -43,10 +42,13 @@ type Props = {
 };
 
 export const SignIn = ({navigation}: Props) => {
-  const deviceCountry = useSelector(deviceCountrySelector);
+  const deviceLocation = useSelector(deviceLocationSelector);
```
Comment (victor-oancea-leftclick-io) [inline src/screens/UserRegistrationFlow/SignIn/index.tsx:45]: Actually, u dont need this either

Comment (pisarevkir) [reply src/screens/UserRegistrationFlow/SignIn/index.tsx:45]: I don't need `deviceLocation`? Why so?

File: src/screens/UserRegistrationFlow/SignIn/index.tsx (hunk 2 above)
Comment (pisarevkir) [reply src/screens/UserRegistrationFlow/SignIn/index.tsx:50]: I open the app on Sign In screen -> I switch the registration method to "Phone", "get device location" method is not completed yet or it returns an error, what country should be selected in the input?

File: src/screens/SettingsFlow/PersonalInformation/index.tsx (hunk 3 above)
Comment (pisarevkir) [reply src/screens/SettingsFlow/PersonalInformation/index.tsx:46]: This is a fallback for the cases when user entity doesn't contain country or that country is not from the `@constants/countries`

File: src/components/ModifyPhoneNumber/index.tsx (hunk 4 above)
Comment (pisarevkir) [reply src/components/ModifyPhoneNumber/index.tsx:38]: https://github.com/ice-blockchain/mobile-app/pull/47#discussion_r938455794

File: src/utils/country.ts (hunk 1 above)
Comment (pisarevkir) [reply src/utils/country.ts:13]: Fallback, so if country in null / is not from the list, we could pre select at least something in the phone input

**Review by victor-oancea-leftclick-io — APPROVED** [review]

File: src/screens/UserRegistrationFlow/SignIn/index.tsx (hunk 2 above)
Comment (victor-oancea-leftclick-io) [reply src/screens/UserRegistrationFlow/SignIn/index.tsx:50]: first one is fine in that case

**Review by victor-oancea-leftclick-io — APPROVED** [review]

**Review by max-tal — APPROVED** [review]

**Review by robertpreoteasa — APPROVED** [review]

**Pull request merged**